src/instrumentations/logging.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Options } from '../options';
+import { Span } from '@opentelemetry/api';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type LogRecord = Record<string, any>;
+
+export function configureLogInjection(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  instrumentation: any,
+  options: Options
+) {
+  if (!options.logInjectionEnabled) {
+    return;
+  }
+
+  if (
+    typeof instrumentation['setConfig'] !== 'function' ||
+    typeof instrumentation['getConfig'] !== 'function'
+  ) {
+    return;
+  }
+
+  const logHook = (_span: Span, record: LogRecord) => {
+    record['service'] = options.serviceName;
```

Comment:
any reason to not have this in the instrumentation packages? Otel spec recommends correlating with resources https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/overview.md#log-correlation

Reply:
AFAIK currently it's not possible to access the provider's resource information, or even reading span attributes; at least I haven't found any.

For reference:
https://open-telemetry.github.io/opentelemetry-js/interfaces/span.html
https://open-telemetry.github.io/opentelemetry-js/interfaces/tracerprovider.html

If that is the case I guess we need changes to the API to read this information, which might be useful for log correlation.

Reply:
Another option is to allow passing a resource directly to the instrumentation constructor. Users can do it manually and our distributions can do it automatically. Issue with this is that a single node project might want to use two different resources to represent sub-components which wouldn't work in this case.

Reply:
Updated it to read from the SDK span resource attributes.